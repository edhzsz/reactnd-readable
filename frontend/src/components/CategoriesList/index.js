import React from 'react';
import PropTypes from 'prop-types';
import './CategoriesList.css'
import Category from '../Category'

const CategoriesList = (props) => {
  return (
    <div className="categories-list">
      <div className="category">All</div>
      <Category name="React" isSelected={true} />
      <Category name="Redux" />
      <Category name="Udacity" />
      <div className="category">+ Add Category</div>
    </div>
  );
}

CategoriesList.propTypes = {

}

export default CategoriesList;