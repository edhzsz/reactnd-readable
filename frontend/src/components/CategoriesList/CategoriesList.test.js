import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import CategoriesList from '.';

const wrapper = shallow(<CategoriesList />);

describe('Component: CategoriesList', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<CategoriesList />, div);
  });

  it('should match its snapshot with required props', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});