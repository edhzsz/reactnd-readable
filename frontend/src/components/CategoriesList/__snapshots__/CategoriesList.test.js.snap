// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: CategoriesList should match its snapshot with required props 1`] = `
<div
  className="categories-list"
>
  <div
    className="category"
  >
    All
  </div>
  <Category
    isSelected={true}
    name="React"
  />
  <Category
    name="Redux"
  />
  <Category
    name="Udacity"
  />
  <div
    className="category"
  >
    + Add Category
  </div>
</div>
`;
